### PR TITLE
Add documentation for background job's state transition

### PR DESCRIPTION
**Pull request by jowlyzhang**
The `SchedulePending*` API is a bit confusing since it doesn't immediately schedule the work and can be confused with the actual scheduling. So I have changed these to be `EnqueuePending*` and added some documentation for the corresponding state transitions of these background work.

Test Plan:
existing tests

### CODE REVIEW

File: db/db_impl/db_impl_compaction_flush.cc [hunk 1]
```diff
@@ -3041,16 +3042,15 @@ bool DBImpl::SchedulePendingFlush(const FlushRequest& flush_req) {
...
   mutex_.AssertHeld();
   if (reject_new_background_jobs_) {
     return;
   }
   if (!cfd->queued_for_compaction() && cfd->NeedsCompaction()) {
-    TEST_SYNC_POINT_CALLBACK("SchedulePendingCompaction::cfd",
+    TEST_SYNC_POINT_CALLBACK("EnqueuePendingCompaction::cfd",
                              static_cast<void*>(cfd));
     AddToCompactionQueue(cfd);
-    ++unscheduled_compactions_;
```
Comment (cbi42) [inline db/db_impl/db_impl_compaction_flush.cc:3053]: Should not remove this?

Comment (jowlyzhang) [reply db/db_impl/db_impl_compaction_flush.cc:3053]: It's moved into `AddToCompactionQueue` since this line of code is duplicated.

**Review by cbi42 — APPROVED** [review]
LGTM thanks

**Comment by jowlyzhang** [comment]
> LGTM thanks

@cbi42 thank you for the quick review!

**Pull request closed without merging**